internal/js/modules/k6/browser/common/frame_manager.go

```diff
@@ -540,7 +540,40 @@ func (m *FrameManager) requestStarted(req *Request) {
 		frame.pendingDocument = &DocumentInfo{documentID: req.documentID, request: req}
 		frame.pendingDocumentMu.Unlock()
 	}
-	m.logger.Debugf("FrameManager:requestStarted", "fmid:%d rurl:%s pdoc:nil", m.ID(), req.URL())
+
+	if !m.page.hasRoutes() {
+		return
+	}
+
+	route := NewRoute(m.logger, m.page.mainFrameSession.networkManager, req)
+	for _, r := range m.page.routes {
+		matcher, err := urlMatcher(r.path, r.jsRegexChecker)
```

Comment:
Could we create this url matcher once in the `route` method in `page` where we create a `NewRoute` instance? At the moment a new instance is created every time a request is created.

Reply:
Great idea! Updated in https://github.com/grafana/k6/pull/4953/commits/48787ab2a9c41c510eed541f1fafba49167c4ecb